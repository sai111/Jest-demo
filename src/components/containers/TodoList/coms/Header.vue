<template>
  <div class="header-wrap">
    <div class="header-wrap-box">
      TodoList
      <input
        type="text"
        v-model="inputValue"
        :placeholder="palceHolder"
        class="input-box"
        data-test="input"
        @keyup.enter="addTodoItem()"
      >
    </div>
  </div>
</template>
<script>
export default {
  name: 'Header',
  components: {},
  props: {},
  data () {
    return {
      palceHolder: '请输入',
      inputValue: ''
    }
  },
  mounted () {},
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>
<style scoped lang="scss">
.header-wrap {
  width: 100%;
  height: 80px;
  line-height: 80px;
  background: #000;
  .header-wrap-box {
    width: 600px;
    margin: 0 auto;
    color: #fff;
    font-size: 24px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    .input-box {
      height: 32px;
      width: 300px;
      outline: none;
      padding: 0 6px;
      box-sizing: border-box;
    }
  }
}
</style>
