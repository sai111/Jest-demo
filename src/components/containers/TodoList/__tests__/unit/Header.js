

import { shallowMount } from '@vue/test-utils'
import Header from '../../coms/Header.vue'
import findTestWrapper from '../../../../../utils/testUtils.js'
/**
 * 1. 测试是否存在input框
 * 2. 初始内容为空
 * 3. 数据动态变化
 * 4. 回车事件: 有输入值、无输入值、事件触发后的效果
 */
describe('Header 组件', () => {
  it(' 样式发生改变，做提示', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })
  it('input框存在', () => {
    const wrapper = shallowMount(Header)
    // const input = wrapper.find('[data-test="input"]')
    const input = findTestWrapper(wrapper, 'input')
    // exists vue/test-utils 检查input(dom)元素是否存在
    expect(input.exists()).toBe(true)
  })
  it('input 框初始内容为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })
  it('input 数据动态变化', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('测试')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('测试')
  })
  
  it('nput 回车事件无内容不通过', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('')
    input.trigger('keyup.enter')
    // 回车会增加一个数据，当没有内容时不应该向外面触发事件
    // toBefalsy() 测试不通过
    // toBeTruthy() 测试通过
    expect(wrapper.emitted().add).toBeFalsy()
  })
  
  it('input 回车事件有内容通过', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('测试')
    input.trigger('keyup.enter')
    // toBefalsy() 测试不通过
    // toBeTruthy() 测试通过
    expect(wrapper.emitted().add).toBeTruthy()
  })
  
  it('input 回车事件,清空inputValue', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('测试')
    input.trigger('keyup.enter')
    // 回车会增加一个数据，当没有内容时不应该向外面触发事件
    // toBefalsy() 测试不通过
    // toBeTruthy() 测试通过
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.$data.inputValue).toBe('')
  })
})
