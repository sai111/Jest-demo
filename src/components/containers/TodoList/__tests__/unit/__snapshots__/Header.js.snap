// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 样式发生改变，做提示 1`] = `
<div class="header-wrap">
  <div class="header-wrap-box">
    TodoList
    <input type="text" placeholder="请输入" data-test="input" class="input-box">
  </div>
</div>
`;

exports[`Header 样式发生改变，做提示 1`] = `
<div class="header-wrap">
  <div class="header-wrap-box">
    TodoList
    <input type="text" placeholder="请输入" data-test="input" class="input-box">
  </div>
</div>
`;

exports[`Header 组件  样式发生改变，做提示 1`] = `
<div class="header-wrap">
  <div class="header-wrap-box">
    TodoList
    <input type="text" placeholder="请输入" data-test="input" class="input-box">
  </div>
</div>
`;
